src/main/java/adris/altoclef/tasks/misc/RepairToolTask.java

```diff
@@ -0,0 +1,197 @@
+            new ItemTarget(Items.WARPED_FUNGUS_ON_A_STICK),
+            new ItemTarget(Items.BOW)
+        );
+    }   
+    @Override
+    protected void onStart(AltoClef mod) {
+        _throwTimer.reset();
+        _toRepair = Arrays.stream(_toRepair).filter(target -> needRepair(mod, target)).toArray(ItemTarget[]::new); //Filter out items type we can't repair
+        Optional<ItemTarget> ItemTargetOPTRepair = Arrays.stream(_toRepair).findFirst(); //Get the first element in our list
+        if (!ItemTargetOPTRepair.isPresent()) {
+            Debug.logWarning("There is nothing to repair");
+        }
+    }
+
+    @Override
+    protected Task onTick(AltoClef mod) {
+        //We start this task by filtering out every item type that we can't repair :
+        //All items without mending or with no damage
+        _toRepair = Arrays.stream(_toRepair).filter(target -> needRepair(mod, target)).toArray(ItemTarget[]::new);
+        
+        //After that, we get the first item type to repair on the list
+        Optional<ItemTarget> ItemTargetOPTRepair = Arrays.stream(_toRepair).findFirst();
+        
+        if (ItemTargetOPTRepair.isPresent()) { //If the list is not empty
+            ItemTarget ItemTargetRepair = ItemTargetOPTRepair.get(); //We get the (real) first item on the list
+            
+            List<Slot> SlotRepair = mod.getItemStorage().getSlotsWithItemPlayerInventory(false, ItemTargetRepair.getMatches()); //And we get a list of every slot with that item
+            
+            Slot SlotRepairTarget = new PlayerSlot(-1); //Placeholder slot, will never get used if not replaced
+            boolean FoundSomethingToRepair = false; //Will be set if we find the slot with the item to repair
+            for (int i = 0; i < SlotRepair.size(); ++i) { //For every item slot that is on our list
+                if (!FoundSomethingToRepair & StorageHelper.getItemStackInSlot(SlotRepair.get(i)).getDamage() != 0) { //if we can repair it
+                    if (haveMending(mod,StorageHelper.getItemStackInSlot(SlotRepair.get(i)))) { //and it have mending
+                        SlotRepairTarget = SlotRepair.get(i); //Replace the placeholder slot with the slot we found
+                        FoundSomethingToRepair = true;
+                    }
+                }
+            }
+            if (FoundSomethingToRepair) { //If we found our slot, we can now repair the item !
+                setDebugState("Repairing "+StorageHelper.getItemStackInSlot(SlotRepairTarget).getName().getString());
+                if (!_throwTimer.elapsed()){ //If we just used a experience bottle, get the item in our hand to repair
+                    mod.getSlotHandler().forceEquipSlot(SlotRepairTarget);
+                    return null;
+                }
+                //Get the nearest experience orb
+                Optional<Entity> expentityOPT = mod.getEntityTracker().getClosestEntity(mod.getPlayer().getPos(), ExperienceOrbEntity.class);
```

Comment:
This is an annoying gotcha, but avoid always gunning for the closest entity/block. There could be a ping-pong pattern of the bot getting stuck between two targets (https://youtu.be/uROEqwyzn3o?t=6946)

Using `DoToClosest ... Task` can fix this, as it checks for this bug.

```java
return new DoToClosestEntityTask(entity -> {
    // Make sure our repair item is equipped when we're about to collect the orb!
    if (entity.isInRange(mod.getPlayer(), 3)) {
        mod.getSlotHandler().forceEquipSlot(SlotRepairTarget);
    }
    return new GetToEntityTask(entity);
});
```

Reply:
Interesting ... maybe this is why my @pickup task likes to whit itself sometimes